src/drivers/fluid_opensles.c

```diff
@@ -0,0 +1,414 @@
+  if (time_delta == 0)
+  {
+    dev->next_expected_enqueue_time += current_time + wait_in_theory;
+  }
+  else
+  {
+    dev->next_expected_enqueue_time += wait_in_theory;
+  }
+  /* take some sleep only if it's running ahead */
+  if (time_delta > 0)
+  {
+    usleep (time_delta);
+  }
+}
+
+void opensles_callback(SLAndroidSimpleBufferQueueItf caller, void *pContext)
+{
+  fluid_opensles_audio_driver_t* dev = (fluid_opensles_audio_driver_t*) pContext;
+  int err;
+  SLresult result;
+
+  process_fluid_buffer(dev);
+
+  if (dev->is_sample_format_float)
+  {
+    result = (*caller)->Enqueue (
+    dev->player_buffer_queue_interface, dev->sles_buffer_float, dev->period_frames * sizeof (float) * NUM_CHANNELS);
+  }
+  else
+  {
+    result = (*caller)->Enqueue (
+    dev->player_buffer_queue_interface, dev->sles_buffer_short, dev->period_frames * sizeof (short) * NUM_CHANNELS);
+  }
+  if (result != 0)
+  {
+    err = result;
+    /* Do not simply break at just one single insufficient buffer. Go on. */
+  }
+}
+
+static fluid_thread_return_t
+fluid_opensles_audio_run(void* d)
+{
+  fluid_opensles_audio_driver_t* dev = (fluid_opensles_audio_driver_t*) d;
+  short *out_short = dev->sles_buffer_short;
+  float *out_float = dev->sles_buffer_float;
+  int period_frames;
+  int err;
+  SLresult result;
+
+  period_frames = dev->period_frames;
+
+  while (dev->cont)
+  {
+    adjust_latency (dev);
+
+    process_fluid_buffer(dev);
+
+    if (dev->is_sample_format_float)
+    {
+      result = (*dev->player_buffer_queue_interface)->Enqueue (
+      dev->player_buffer_queue_interface, out_float, period_frames * sizeof (float) * NUM_CHANNELS);
+    }
+    else
+    {
+      result = (*dev->player_buffer_queue_interface)->Enqueue (
+      dev->player_buffer_queue_interface, out_short, period_frames * sizeof (short) * NUM_CHANNELS);
+    }
+    if (result != 0)
+    {
+      err = result;
+      /* Do not simply break at just one single insufficient buffer. Go on. */
+    }
+  }	/* while (dev->cont) */
+
+  return NULL;
+}
+
+void process_fluid_buffer(fluid_opensles_audio_driver_t* dev)
+{
+  short *out_short = dev->sles_buffer_short;
+  float *out_float = dev->sles_buffer_float;
+  int period_frames = dev->period_frames;
+  float *float_callback_buffers[2];
+  short *short_callback_buffers[2];
+  
+  if (dev->is_sample_format_float)
+  {
+    if (dev->callback)
+    {
+      float_callback_buffers [0] = out_float;
+      float_callback_buffers [1] = out_float;
+
+      (*dev->callback)(dev->synth, period_frames, 0, NULL, 2, float_callback_buffers);
+    }
+    else
+    {
+      fluid_synth_write_float(dev->synth, period_frames, out_float, 0, 2, out_float, 1, 2);
+    }
+  }
+  else
+  {
+    if (dev->callback)
+    {
+      short_callback_buffers [0] = out_short;
+      short_callback_buffers [1] = out_short;
+
+      (*dev->callback)(dev->synth, period_frames, 0, NULL, 2, (float**) short_callback_buffers);
```

Comment:
Same as with oboe: shorts aren't supported for custom callbacks. You shouldn't attempt to support it.

Reply:
Alright. The entire function is being removed anyways.